app/js/trouble.js

```diff
@@ -1,123 +1,141 @@
-				open();
-			} else {
-				function getStatus() {
-					return
-					{
-						status: "open"
-					}
-				}
-				close();
-			}
-			if (controlsMegaMenu) {
-				closeMegaMenu();
-
-		}
-
-		$dropdown.on(transitionEnd, function () {
-			if ($dropdown.data('status') == "open") {
-				$dropdown.addClass('no_transition');
-				$dropdown.css('height', 'auto');
-			}
-		});
-
-		$toggle.on('click', toggleDropdown);
-		this.open = open;
-		this.close = close;
-		this.setDropdownHeight = setDropdownHeight;
-	};
-
-}(window.gbl || {}, jQuery));
\ No newline at end of file
+    (function (gbl, $) {
+
+        gbl.dropdown = function (options) {
+
+            var $dropdown = options.dropdown;
+            var $toggle = options.toggleButton;
+            var token = +new Date();
+            var ns = options.namespace;
+            var elList = $('.dropdowns');
+            var controlsMegaMenu = (options.controlsMegaMenu && options.controlsMegaMenu === false)? false: true;
+            var clickAnywhereToClose = (options.clickAnywhereToClose && options.clickAnywhereToClose === false) ? false: true;
+                var transitionEnd = gbl.utilities.whichTransitionEvent();
+
+
+            $dropdown.addClass('gbl_dropdown').data('status', 'closed');
+            $toggle.addClass('gbl_dropdown_trigger');
+            $dropdown.attr('aria-expanded', 'false');
+            $toggle.attr('aria-controls', $dropdown.attr('id'));
+            $dropdown.wrapInner('<div class="measureHeight"></div>');
+
+            function closeMegaMenu() {
+                if (gbl.megaMenu && gbl.megaMenu.small && typeof gbl.megaMenu.small.close == 'function') {
+                    gbl.megaMenu.small.close();
+                }
+            }
+
+            function setCloseHandler() {
+                $(document).on('click.' + ns, function (e) {
+                    var $clicked = $(e.target);
+                    if (!$clicked.is($dropdown) && ($clicked.parents().filter($dropdown).length === 0)) {
+                        close();
+                    }
+                });
+            }
+
+
+
+            function removeCloseHandler() {
+                $(document).off('click.' + ns);
+            }
+
+
+
+            function processDropdownEls() {
+                var list = Array.prototype.slice.call(elList);
+            }
+
+
+
+            function setDropdownHeight() {
+                $dropdown.height($dropdown.find('.measureHeight').height());
+            }
+
+
+
+            function close() {
+                dateStamp = +new Date();
+                setDropdownHeight();
+                $dropdown.data("status", 'closed');
+                $dropdown.removeClass('gbl_dropdown_active');
+
+                Array.prototype.foreach = function(cb) {
+                    for (var i = 0; i < this.length; ++i) {
+                        cb(this[i]);
+                    }
+                };
+
+                setTimeout(function() { setDropdownHeight(); }, 1000);

```

Comment:
While this removes "making" a function in a loop, the desire here is to address it in one of two ways: 1. remove the anonymous function expression and either assign it to a variable or declare the function outside the loop, or 2. remove the infinite while loop entirely. By invoking the function here, it will run immediately when the code executes rather than when the timeout event fires later.